pkg/kubelet/container/container_hash_test.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package container
+
+import (
+	"encoding/json"
+	"hash/fnv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/kubernetes/pkg/kubelet/container/testing/v115"
+	"k8s.io/kubernetes/pkg/kubelet/container/testing/v116"
+	"k8s.io/kubernetes/pkg/kubelet/container/testing/v1xx"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+)
+
+// hashContainerV1xx returns the hash of the container. It is used to compare
+// the running container with its desired spec.
+func hashContainerV1xx(_ *testing.T, container *v1xx.Container) uint64 {
+	hash := fnv.New32a()
+	// Omit nil or empty field when calculating hash value
+	// Please see https://github.com/kubernetes/kubernetes/issues/53644
+	containerJson, _ := json.Marshal(container)
+	hashutil.DeepHashObject(hash, containerJson)
+	return uint64(hash.Sum32())
+}
+
+// hashContainerV116 returns the hash of the container. It is used to compare
+// the running container with its desired spec.
+func hashContainerV116(_ *testing.T, container *v116.Container) uint64 {
```

Comment:
after this PR is committed, the hash for a given container spec should not change. I would put the JSON blob in a package variable in this package, then in the test, unmarshal it to a container object, compute the hash, and compare it to recorded hashes computed against 1.15/1.14 containers